controller/src/main/java/org/jboss/as/controller/parsing/ParseUtils.java

```diff
@@ -565,8 +565,8 @@ public static ModelNode parseAttributeValue(final String value, final boolean is
         final String trimmed = value == null ? null : value.trim();
         ModelNode node;
         if (trimmed != null) {
-            if (isExpressionAllowed && isExpression(trimmed)) {
-                node = new ModelNode(new ValueExpression(trimmed));
+            if (isExpressionAllowed && isExpression(value)) {
+                node = new ModelNode(new ValueExpression(value));
```

Comment:
This should do the same test that the non-expression else block does, and use 'trimmed' unless attributeType is STRING or PROPERTY. An expression that's meant to resolve to a string convertible to some other type shouldn't have leading/trailing whitespace and we shouldn't stop being helpful about that.

Reply:
Ok, done...